src/js/playlist/source.ts

```diff
@@ -10,14 +11,39 @@ import { trim, extension } from 'utils/strings';
  * @property {string} label - The quality label to be used with multiple mp4/webm sources.
  */
 
-const Source = function(config) {
+export type PlaylistItemSource = {
+    type: string;
+    default: boolean;
+} & PlaylistItemConfig;
```

Comment:
Does this override work? I tried something similar, but did not see type checks enforced on these fields. I had to use `Omit` to remove optional fields so that the one's we're making required are treated that way.

Reply:
```ts
// OK: type-check ensured `default` is a boolean / required
export type PlaylistFeedSource = Omit<PlaylistMRSSSource, 'default'> & {
    default: boolean;
};

// NO: `default` is still optional based on `PlaylistMRSSSource` in both these examples
export type PlaylistFeedSource = PlaylistMRSSSource & {
    default: boolean;
};

export type PlaylistFeedSource = {
    default: boolean;
} & PlaylistMRSSSource;
```

Reply:
i'll use omit to be safe but i found my ide throwing an error on a `PlaylistItemSource` when either of these props were not preset